### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by sritej-reddy**

### CODE REVIEW

File: src/count.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	}
+	if (num2 == 0)
+		return 0;
+	if (num1 > num2)
+		return 1;
+	for (int i = num1; i <= num2; i++)
+	{
+		if (i%num1 == 0)
+			count++;
+	}
```
Comment (krushiraj) [inline src/count.cpp]: Can you think of any solution without using iterations and make this an O(1) solution?

File: src/countFactorials.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	long i=0;
+	while(1)
+	{
+		if(fact(i)>=l && fact(i)<=h)
+		count++;
+		if (fact(i) > h)
+			break;
+		i++;
+	}
+
```
Comment (krushiraj) [inline src/countFactorials.cpp:54]: Can you discuss the drawbacks of this algo and what can be done to fix it?

File: src/countOnes.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Sample Input : 3
+Output : 3 (111 is the multiple )
+*/
+
+int countOnes(int number)
+{
+	if (number==0)
+		return -1;
+	unsigned long i=111;
```
Comment (krushiraj) [inline src/countOnes.cpp:29]: did you check for the overflow conditions here?

File: src/decimalToOctal.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		return 0;
+	int octal=0,count=0,temp=0;
+	while(num>0)
+	{
+		temp = num % 8;
+		octal = octal + (temp*pow(10, count));
+		num /= 8;
+		count++;
+	}
+
```
Comment (krushiraj) [inline src/decimalToOctal.cpp:33]: So here pow is being called `the no of digits in num times`. Can you make of use of pow only once and get the solution done by having a lesser cost operations like math operations?

File: src/isOlder.cpp [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+				else return 0;
+			}
+		}
+
+	}
+	else
+		return -1;
+}
+
+
```
Comment (krushiraj) [inline src/isOlder.cpp:131]: Re-write the whole thing by breaking it into functions and make it more modular and easy to look and understand.

File: src/isPerfect.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+*/
+
+#include<stdio.h>
+
+bool check(long num)
+{
+	int sum = 0;
+	int i;
+	for (i = 1; i <= num / 2; i++)
+	{
```
Comment (krushiraj) [inline src/isPerfect.cpp:26]: Good optimization strategy by checking upto `num/2`

File: src/nthNumber.cpp [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		 nth prime number for nthPrime()
+
+ERROR CASES: return -1 for the error cases
+*/
+bool isprime(int num)
+{
+	if (num == 2)
+		return true;
+	for (int i = 2; i <= num / 2 + 1; i++)
+	{
```
Comment (krushiraj) [inline src/nthNumber.cpp:23]: All primes are odds, can you make use of this fact and optimize the function to some more extent?

File: src/octalToHexadecimal.cpp [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+		octtemp = num % 10;
+		dec += octtemp*pow(8, i);
+		num /= 10;
+	}
+	int count = 0,hextemp=0,hex=0;
+	while (dec)
+	{
+		hextemp = dec % 16;
+		hex = hex + (hextemp*pow(10, count));
+		dec /= 16;
```
Comment (krushiraj) [inline src/octalToHexadecimal.cpp:42]: Same here, try to make use of pow less no of times.

File: src/octalToHexadecimal.cpp [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+		}
+		else
+			*(hex++) = (char)t;
+
+
+	}
+	rt = hex--;
+	//as test case doesn't contain which gives char type of hex number as o/p so i am passing (int)
+	return (int)hex;*/
+	return hex;
```
Comment (krushiraj) [inline src/octalToHexadecimal.cpp:84]: Good to see that you tried this for >10.

File: src/swapBits.cpp [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+	while (number >0)
+	{
+		bin += ((number%2)*pow(10, size++));
+		number /= 2;
+	}
+	if (p2 + n > size)
+	{
+			size = p2 + n;
+	}
+
```
Comment (krushiraj) [inline src/swapBits.cpp:119]: I guess you can do it in simpler algo. 

HINT: try to make use of operators in C that work on bits of numbers.

**Review by krushiraj — CHANGES_REQUESTED** [review]
Please commit with the requested changes.

File: src/countOnes.cpp (hunk 3 above)
Comment (sritej-reddy) [reply src/countOnes.cpp:29]: yes it is overflowing..manoj said he asked you about it and you said to him that you will check the question ones so i ignored that exception

File: src/nthNumber.cpp (hunk 7 above)
Comment (sritej-reddy) [reply src/nthNumber.cpp:23]: ohk done

File: src/countOnes.cpp (hunk 3 above)
Comment (sritej-reddy) [reply src/countOnes.cpp:29]: anna i got a solution for it now..it may work

Comment (sritej-reddy) [reply src/countOnes.cpp:29]: vacchindi anna

File: src/countFactorials.cpp (hunk 2 above)
Comment (sritej-reddy) [reply src/countFactorials.cpp:54]: in every iteration factorial is found, it is a time taking process, so to solve this we can compute factorial in other way (storing a number and multiplying it with next number in the next iteration, as finding the factorials in this problem is of consequence numbers)

Comment (sritej-reddy) [reply src/countFactorials.cpp:54]: i will make required changes and commit it to master branch

**Pull request closed without merging**